### PR TITLE
Convert createActionManager

### PR DESCRIPTION
**Pull request by kebeclibre**

### CODE REVIEW

File: addons/web/static/tests/helpers/mock_server.js [hunk 1]
```diff
@@ -506,6 +508,22 @@ class MockServer {
     return result;
   }
 
+  mockNameSearch(model, args, _kwargs) {
+    var str = args && typeof args[0] === 'string' ? args[0] : _kwargs.name;
```
Comment (aab-odoo) [inline addons/web/static/tests/helpers/mock_server.js]: what is var?

File: addons/web/static/tests/legacy/control_panel/search_bar_tests.js [hunk 2]
```diff
@@ -54,127 +61,123 @@ odoo.define('web.search_bar_tests', function (require) {
...
+                    searchReadCount++;
+                }
+            };
+            const webClient = await createWebClient({ testConfig , mockRPC });
+            await doAction(webClient, 1);
+
+            // Where to put that ? either in tests infra, or in business legacy compatibility layer ?
+            // session: {
+            //     getTZOffset() {
+            //         return 360;
```
Comment (aab-odoo) [inline addons/web/static/tests/legacy/control_panel/search_bar_tests.js]: it wasn't necessary? you can patchAndCleanup the session I guess

File: addons/web/static/tests/legacy/control_panel/search_bar_tests.js [hunk 3]
```diff
@@ -319,93 +315,81 @@ odoo.define('web.search_bar_tests', function (require) {
...
 
-        QUnit.skip('update suggested filters in autocomplete menu with Japanese IME', async function (assert) {
+        QUnit.test('update suggested filters in autocomplete menu with Japanese IME', async function (assert) {
             assert.expect(4);
 
             // The goal here is to simulate as many events happening during an IME
             // assisted composition session as possible. Some of these events are
             // not handled but are triggered to ensure they do not interfere.
             const TEST = "TEST";
             const テスト = "テスト";
```
Comment (aab-odoo) [inline addons/web/static/tests/legacy/control_panel/search_bar_tests.js]: lol

File: addons/web/static/src/legacy/utils.js [hunk 4]
```diff
@@ -71,6 +72,37 @@ export function makeLegacySessionService(legacyEnv, session) {
...
+  return {database, parameters};
+}
+
+export function makeLegacyTranslationService(legacyEnv, session, legacyTranslatefn) {
+  async function loadTrans(modules, lang, url, load=false) {
+    const bundle = await (load ? loadTranslations(modules, lang, url) : loadTranslations.prom);
+    legacyTranslatefn.database.set_bundle(bundle);
+  }
+  return {
+    name: "legacy_translate",
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/utils.js]: not necessary anymore (?)

File: addons/web/static/tests/helpers/mock_services.js [hunk 5]
```diff
@@ -28,7 +28,8 @@ export const defaultLocalization = {
 };
 
 export function makeFakeLocalizationService(config) {
-  patchWithCleanup(localization, defaultLocalization);
+  const defaultLoc = Object.assign({}, defaultLocalization, config.localizationParameters);
+  patchWithCleanup(localization, defaultLoc);
 
   return {
     name: "localization",
```
Comment (aab-odoo) [inline addons/web/static/tests/helpers/mock_services.js]: remove (everywhere as well) ?

File: addons/web/static/tests/legacy/views/form_tests.js [hunk 6]
```diff
@@ -9725,205 +9721,210 @@ QUnit.module('Views', {
...
+            if (args.method === 'write') {
+                assert.deepEqual(args.args, [
+                    [1],
+                    { name: "aaa" },
+                ]);
+            }
+        };
+
+        const webClient = await createWebClient({ testConfig , mockRPC });
+        addJqueryAbility(webClient);
```
Comment (aab-odoo) [inline addons/web/static/tests/legacy/views/form_tests.js]: remove ? sorry

File: addons/web/static/src/legacy/action_adapters.js [hunk 7]
```diff
@@ -138,6 +138,19 @@ class ActionAdapter extends ComponentAdapter {
   canBeRemoved() {
     return this.__widget.canBeRemoved();
   }
+
+  // We do want to signal willUnmount to the widget and its children
+  // without prejudice to exportState's comment.
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/action_adapters.js]:
```suggestion
  /**
   * The ComponentAdapter calls `on_attach_callback` on `this.widget`. However,
   * in the ActionAdapter, we unset `this.widget` (see @exportState) to keep the child
   * widget alive when this component is unmounted but the controller is still in the stack.
   * This override ensures that `this.widget` is set when willUnmount is called on the
   * ComponentAdapter.
   *
   * @override
   */
```

File: addons/web/static/src/legacy/utils.js [hunk 8]
```diff
@@ -128,6 +129,36 @@ export function makeLegacySessionService(legacyEnv, session) {
   };
 }
 
+export function mapLocalizationParameters(localizationParameters) {
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/utils.js]: only used in tests, right? if so, should not be here

File: addons/web/static/src/legacy/legacy_setup.js [hunk 9]
```diff
@@ -32,8 +34,13 @@ export const legacySetupProm = new Promise((resolve) => {
   // legacy env on the bus in the wowl env
   const legacyRpcService = makeLegacyRpcService(legacyEnv);
   serviceRegistry.add(legacyRpcService.name, legacyRpcService);
+
   const legacySessionService = makeLegacySessionService(legacyEnv, session);
   serviceRegistry.add(legacySessionService.name, legacySessionService);
+
+  const LegacyTranslationService = makeLegacyTranslationService(legacyEnv, session, legacyTranslate._t);
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/legacy_setup.js]: legacyTranslationService

File: addons/web/static/src/legacy/utils.js [hunk 10]
```diff
@@ -128,6 +129,36 @@ export function makeLegacySessionService(legacyEnv, session) {
...
+
+export function makeLegacyTranslationService(legacyEnv, session, legacyTranslatefn) {
+  async function loadTrans(modules, lang, url, load=false) {
+    const bundle = await (load ? loadTranslations(modules, lang, url) : loadTranslations.prom);
+    legacyTranslatefn.database.set_bundle(bundle);
+  }
+  return {
+    dependencies: ["localization"],
+    deploy(env) {
+      legacyTranslatefn.database.load_translations = (modules, lang, url) => {
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/utils.js]: are we sure that this is patched before being called?

File: addons/web/static/src/legacy/utils.js [hunk 11]
```diff
@@ -128,6 +129,36 @@ export function makeLegacySessionService(legacyEnv, session) {
...
+        return loadTrans(modules, lang, url);
+      };
+
+      session.is_bound.then(() => {
+        legacyTranslatefn.database.load_translations = (modules, lang, url) => {
+          return loadTrans(modules, lang, url, true);
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/utils.js]: do we really want to let the legacy code fetch the translations?

File: addons/web/static/src/localization/translation.js [hunk 12]
```diff
@@ -15,6 +18,40 @@ export function _lt(str) {
...
+      throw new Error("Error while fetching translations");
+    }
+    return response.json();
+  };
+}
+
+let loadTransFn;
+export function loadTranslations(modules, lang) {
+  if (!loadTransFn) {
+    loadTransFn = makeLoadTranslations();
```
Comment (aab-odoo) [inline addons/web/static/src/localization/translation.js]: i'm not sure this really needs to be so complicated

**Review by aab-odoo — COMMENTED** [review]
This PR is globally OK, except for the translation part. I'm not sure to understand everything, and I believe it is unnecessary complicated. Let's talk about it tomorrow.

File: addons/web/static/src/legacy/utils.js (hunk 11 above)
Comment (aab-odoo) [reply addons/web/static/src/legacy/utils.js]: yes

File: addons/board/static/tests/dashboard_tests.js [hunk 13]
```diff
@@ -696,191 +704,206 @@ QUnit.test('dashboard intercepts custom events triggered by sub controllers', as
...
     // Add it to dashboard
-    await cpHelpers.toggleFavoriteMenu(actionManager);
-    await testUtils.dom.click(actionManager.el.querySelector('.o_add_to_board > button'));
-    await testUtils.dom.click(actionManager.el.querySelector('.o_add_to_board div button'));
+    await cpHelpers.toggleFavoriteMenu(webClient);
+    await testUtils.dom.click(webClient.el.querySelector(".o_add_to_board > button"));
+    await testUtils.dom.click(webClient.el.querySelector(".o_add_to_board div button"));
 
-    actionManager.destroy();
+    webClient.destroy();
```
Comment (aab-odoo) [inline addons/board/static/tests/dashboard_tests.js]: no need to destroy (done via registerCleanup in createWebClient)

File: odoo/addons/base/static/tests/base_settings_tests.js [hunk 14]
```diff
@@ -1,445 +1,560 @@
-odoo.define('base.settings_tests', function (require) {
```
Comment (aab-odoo) [inline odoo/addons/base/static/tests/base_settings_tests.js]: could you try not to prettify all this file?

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by kebeclibre** [comment]
robodoo rebase-ff

**Comment by kebeclibre** [comment]
robodoo retry

**Pull request closed without merging**